src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -321,22 +307,21 @@ void GIVEN_managedCAConfiguration_WHEN_updatedToCustomCAConfiguration_THEN_serve
         doReturn(chain).when(certificateStoreSpy).loadCaCertificateChain(privateKeyUri, certificateUri);
 
         AtomicReference<CertificateUpdateEvent> eventRef = new AtomicReference<>();
-        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> asyncCall =
-                TestUtils.asyncAssertOnConsumer(eventRef::set, 2);
         GetCertificateRequest request = buildCertificateUpdateRequest(
-                GetCertificateRequestOptions.CertificateType.SERVER, asyncCall.getRight());
+                GetCertificateRequestOptions.CertificateType.SERVER, eventRef::set);
 
         givenNucleusRunningWithConfig("config.yaml");
         subscribeToCertificateUpdates(request);
         givenCDAWithCustomCertificateAuthority(privateKeyUri, certificateUri);
 
-        // Called 2 times. 1 for initial managed CA and then after the config is changes to use custom CA
-        asyncCall.getLeft().get(5, TimeUnit.SECONDS);
-        CertificateUpdateEvent event = eventRef.get();
-        X509Certificate issuedClientCertificate = event.getCertificate();
-        assertTrue(
-            CertificateTestHelpers.wasCertificateIssuedBy(intermediateCA, issuedClientCertificate), String.format(
-                "Certificate not was not issued by intermediate CA %s", CertificateHelper.toPem(issuedClientCertificate)
-        ));
+        TestHelpers.eventuallyTrue(() -> {
+            try {
+                X509Certificate intermediateCA =  chain[0];
+                X509Certificate issuedClientCertificate = eventRef.get().getCertificate();
+                return CertificateTestHelpers.wasCertificateIssuedBy(intermediateCA, issuedClientCertificate);
+            } catch (CertificateException e) {
+                return  false;
+            }
```

Comment:
Do we want to force the supplier to catch exceptions, or can we handle that in the utility method? 

BTW, I love this. Not counting the number of callbacks is a big win